### PR TITLE
Cv files are uploaded

### PR DESCRIPTION
**Pull request by gabichelsea**
Trello card link:
[https://trello.com/c/vdKazZQB](https://trello.com/c/vdKazZQB)

### CODE REVIEW

File: web/src/main/java/com/epam/rft/atsy/web/handler/FileHandler.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.epam.rft.atsy.web.handler;
+
+
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/handler/FileHandler.java]: Duplicated empty line.

File: web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.epam.rft.atsy.web.handler;
+
+
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java]: Duplicated empty line.

File: web/src/main/java/com/epam/rft/atsy/web/handler/FileHandler.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import java.io.File;
+
+@Component
+public class FileHandler {
+
+  public File getFileByParentDirectoryPathAndFolderNameAndFilename(String parentDirectoryPath,
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/handler/FileHandler.java:22]: Please add a little java doc for this method.

File: web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import java.io.File;
+import java.io.IOException;
+
+@Component
+public class FolderHandler {
+
+  public void createFolderInParentDirectoryPath(String folderName, String parentDirectoryPath)
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java:22]: Please add a little java doc for this method.

File: web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      throws IOException {
+    Assert.notNull(parentDirectoryPath);
+    Assert.notNull(folderName);
+    FileUtils.forceMkdir(new File(parentDirectoryPath + File.separator + folderName));
+  }
+
+  public boolean existFolderNameInParentDirectoryPath(String folderName, String parentDirectoryPath)
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/handler/FolderHandler.java:37]: Please add a little java doc for this method.

File: web/src/test/java/com/epam/rft/atsy/web/handler/FileHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package com.epam.rft.atsy.web.handler;
+
+
```
Comment (lamfalusy) [inline web/src/test/java/com/epam/rft/atsy/web/handler/FileHandlerTest.java]: Duplicated empty line.

File: web/src/test/java/com/epam/rft/atsy/web/handler/FileHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.junit.Test;
+
+import java.io.File;
+
+public class FileHandlerTest extends AbstractHandlerTest {
```
Comment (lamfalusy) [inline web/src/test/java/com/epam/rft/atsy/web/handler/FileHandlerTest.java:10]: Use Given-When-Then comments in all test methods as usual.

File: web/src/test/java/com/epam/rft/atsy/web/handler/FolderHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+public class FolderHandlerTest extends AbstractHandlerTest {
```
Comment (lamfalusy) [inline web/src/test/java/com/epam/rft/atsy/web/handler/FolderHandlerTest.java:14]: Use Given-When-Then comments in all test methods as usual.

File: web/src/main/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateController.java [hunk 9]
```diff
@@ -57,16 +60,17 @@ public ResponseEntity saveOrUpdate(@Valid @RequestBody CandidateDTO candidateDTO
...
+  public ResponseEntity validateCandidate(@Valid @RequestBody CandidateDTO candidateDTO, BindingResult result,
+                                          Locale locale) {
     if (result.hasErrors()) {
       return new ResponseEntity(parseValidationErrors(result.getFieldErrors(), locale), HttpStatus.BAD_REQUEST);
     }
-    if (candidateService.getCandidateDtoByPhone(candidateDTO.getPhone()) != null
-        || candidateService.getCandidateDtoByEmail(candidateDTO.getEmail()) != null) {
-      String errorMessage = messageKeyResolver.resolveMessageOrDefault(CANDIDATE_ERROR_DUPLICATE_MESSAGE_KEY);
-      return new ResponseEntity(new RestResponse(errorMessage), HttpStatus.BAD_REQUEST);
+    // Create a new candidate
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateController.java]: I am not sure what this comment exactly means.

File: web/src/main/java/com/epam/rft/atsy/web/helper/CandidateValidatorHelper.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
+package com.epam.rft.atsy.web.helper;
+
+
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/helper/CandidateValidatorHelper.java]: Duplicated empty line.

File: web/src/main/java/com/epam/rft/atsy/web/helper/CandidateValidatorHelper.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+import com.epam.rft.atsy.web.messageresolution.MessageKeyResolver;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+@Component
+public class CandidateValidatorHelper {
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/helper/CandidateValidatorHelper.java:14]: Please add javadoc to this class.

File: web/src/main/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateController.java (hunk 9 above)
Comment (gabichelsea) [inline web/src/main/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateController.java]: Why not?:DD

File: web/src/main/webapp/resources/js/atsy-statehistory-create.js [hunk 12]
```diff
@@ -58,6 +64,15 @@ function StateHistoryModel() {
...
 
+    self.fileErrorMessage = ko.pureComputed(function() {
+        if (self.fileErrorResponse() === null) {
+            return "";
+        }
+
+        return self.fileErrorResponse().errorMessage;
+    });
+
+
```
Comment (lamfalusy) [inline web/src/main/webapp/resources/js/atsy-statehistory-create.js]: Duplicated empty line.

File: web/src/test/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateControllerTest.java [hunk 13]
```diff
@@ -296,12 +321,217 @@ public void saveOrUpdateShouldRespondWithErrorResponseWhenLanguageSkillIsTooHigh
...
+        .andExpect(status().isOk())
+        .andExpect(jsonPath(JSON_PATH_ERROR_MESSAGE).isEmpty());
+
+    then(this.candidateValidatorHelper).should().isValidExistingCandidate(candidateDTO);
+    then(this.candidateValidatorHelper).should(never())
+        .isValidNonExistingCandidate(any(CandidateDTO.class));
+    verifyZeroInteractions(this.messageKeyResolver);
+  }
+
+
```
Comment (lamfalusy) [inline web/src/test/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateControllerTest.java]: Duplicated empty line.

**Comment by lamfalusy** [comment]
There is a private field in com.epam.rft.atsy.web.controllers.rest.FileUploadController called CANDIDATE_ALREADY_HAS_CV_FILE_MESSAGE_KEY, which is never used.

**Comment by lamfalusy** [comment]
There is a big hole in the form:
image

**Pull request merged**